rest/handle.go

```diff
@@ -0,0 +1,182 @@
+
+	RequestBody() openapi3.RequestBodyOrRef
+	Responses() openapi3.Responses
+}
+
+type securityScheme struct {
+	name   string
+	scheme openapi3.SecurityScheme
+}
+
+// OperationOptions holds configuration for an HTTP operation registered with [Handle].
+// This includes security schemes, parameter definitions, request transformations,
+// and error handling.
+type OperationOptions struct {
+	securityScheme *securityScheme
+	parameters     []openapi3.ParameterOrRef
+	transforms     []func(*http.Request) (*http.Request, error)
+	errHandler     ErrorHandler
+}
+
+// OperationOption configures an operation created by [Handle].
+// Common implementations include parameter validators ([Header], [QueryParam], etc.)
+// and [OnError] for custom error handling.
+type OperationOption func(*OperationOptions)
+
+// OnError configures a custom [ErrorHandler] for an operation.
+// If not specified, operations use a default error handler that logs errors
+// and returns appropriate HTTP status codes.
+//
+// Example:
+//
+//	customErrorHandler := rest.ErrorHandlerFunc(func(ctx context.Context, w http.ResponseWriter, err error) {
+//	    log.Printf("Error: %v", err)
+//	    w.WriteHeader(http.StatusInternalServerError)
+//	    json.NewEncoder(w).Encode(map[string]string{"error": err.Error()})
+//	})
+//	rest.Handle(http.MethodGet, rest.BasePath("/users"), handler, rest.OnError(customErrorHandler))
+func OnError(eh ErrorHandler) OperationOption {
+	return func(oo *OperationOptions) {
+		oo.errHandler = eh
+	}
+}
+
+type operationHandler struct {
+	tracer     trace.Tracer
+	errHandler ErrorHandler
+	transforms []func(*http.Request) (*http.Request, error)
+	inner      http.Handler
+}
+
+// Handle registers an HTTP operation (endpoint) with an [Api].
+//
+// It creates an [ApiOption] that configures both the HTTP routing and the
+// OpenAPI specification for the operation. The operation is registered with
+// the specified HTTP method and path, and handles requests using the provided handler.
+//
+// Parameters:
+//   - method: HTTP method (e.g., http.MethodGet, http.MethodPost)
+//   - path: URL path created with [BasePath] and path building methods
+//   - h: Handler that processes requests and provides OpenAPI schema info
+//   - opts: Optional configuration (parameters, auth, error handling)
+//
+// Example:
+//
+//	getUser := rest.Handle(
+//	    http.MethodGet,
+//	    rest.BasePath("/users").Param("id"),
+//	    userHandler,
+//	    rest.QueryParam("format", rest.Required()),
+//	    rest.Header("Authorization", rest.Required(), rest.JWTAuth("jwt")),
+//	)
+//	api := rest.NewApi("User API", "v1.0.0", getUser)
+func Handle(method string, path Path, h Handler, opts ...OperationOption) ApiOption {
+	return apiOptionFunc(func(ao *ApiOptions) {
+		oo := &OperationOptions{
+			errHandler: defaultErrorHandler(humus.LogHandler("rest")),
+		}
+		for _, opt := range opts {
+			opt(oo)
+		}
+
+		var op openapi3.Operation
+
+		withRequestBody, ok := h.(interface {
```

Comment:
oh this is unnecessary since `h` is a `Handler` whose definition explicitly includes `RequestBody()`

Reply:
addressed locally